app/src/main/java/com/example/taskyy/data/remote/Workers/AgendaItemWorker.kt

```diff
@@ -37,40 +36,64 @@ class AgendaItemWorker(
                 when (reminder.action) {
                     AgendaItemAction.CREATE -> {
                         try {
-                            taskyyApi.createReminder(reminder.toReminderDTO())
+                            taskyyApi.createReminder(
+                                taskyyDatabase.agendaDao().getReminderByEventId(reminder.id)
+                                    .toReminderDTO()
+                            )
+                            taskyyDatabase.pendingReminderRetryDao().removePendingReminder(reminder)
```

Comment:
Even though very unlikely, you'll need to take some care when launching two such suspend functions one after another when either both have to be executed or none. Imagine, you'd finish the first call to `taskyyApi.createReminder` and then before the code reaches the next suspend function, the coroutine is cancelled. This could happen when this suspend fun is launched in `viewModelScope` for example and the user navigates back. In that case the line to update the local DB won't be executed, however the API was already notified about that change. So the app still thinks the reminder is pending and tries to sync it the next time, even though it is already synced (and this could lead to invalid states, since your invalid sync could override the reminder on the API which could've been modified already at this point).

The fix is to make sure that the second suspend function is not cancellable. The best way to do this is with a custom application wide coroutine scope, which lives as long as your app's process:
```
class MyApp: Application() {
    val applicationScope = CoroutineScope(SupervisorJob())
    
    // ...
}
```
You can then provide an instance of this scope in your Dagger module:
```
fun provideApplicationScope(app: Application): CoroutineScope {
    return (app as MyApp).applicationScope
}
```
Then, inject it into the class you need it in:
```
class MyRepository(
    private val applicationScope: CoroutineScope,
    ...
)
```
And make sure the second suspend function is executed in this new scope which won't be cancelled:
```
applicationScope.launch {
    // DB call
}.join()
```
This way, if the coroutine reached this point and wasn't cancelled, we make sure that it also won't be cancelled during this second call. The `join()` makes sure to still suspend the outer suspend function until the DB call finished.

Reply:
Does this go in the App class? This one?

@HiltAndroidApp
public class App extends Application {
}

When I tried to put it there it gave me an error so I don't think that's where it is supposed to be